File: api/src/main/java/com/greenhome/api/model/apartment/Apartment.java
package com.greenhome.api.model.apartment;

import com.greenhome.api.model.building.Building;
import com.greenhome.api.model.property.Property;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Data
@Entity
@Table(name = "apartment")
public class Apartment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    
    private String name;
    
    @JoinColumn(name = "property_id")
    @ManyToOne
    private Building property;
    
    private double area;
    
    private short bedroomCount;
    
    private short bathroomCount;
    
    private boolean hasGarage;
    
    private boolean hasServiceArea;
    
    private boolean hasBalcony;
    
    private boolean hasBarbecueGrill;
    
    private boolean hasCloset;
}

File: front/src/types/apartments/apartment.ts
export default interface Apartment {
  id?: number;
  name: string;
  propertyId: number;
  area: number;
  bedroomCount: number;
  bathroomCount: number;
  hasGarage: boolean;
  hasServiceArea: boolean;
  hasBalcony: boolean;
  hasBarbecueGrill: boolean;
  hasCloset: boolean;
}
